tck/src/main/java/ee/jakarta/tck/concurrent/spec/Platform/virtual/VirtualThreadServlet.java

```diff
@@ -0,0 +1,468 @@
+                    (Runnable) new LookupActionCaptureThread(results,
+                            "java:app/concurrent/ManagedScheduledExecutorDDVirtual"),
+                    TestConstants.pollInterval.toMillis(), TimeUnit.MILLISECONDS);
+    
+            assertTimeoutPreemptively(TestConstants.waitTimeout, () -> {
+                for (; 2 != results.size(); Wait.sleep(TestConstants.pollInterval)) {
+                    // Wait till we have exactly 2 results indicating 1 run of schedule
+                }
+            });
+    
+            assertTrue(oneTimeFuture.isDone());
+            assertEquals(2, results.size());
+    
+            Thread thread = (Thread) results.poll(1, TimeUnit.MILLISECONDS);
+            Object result = results.poll(1, TimeUnit.MILLISECONDS);
+    
+            if (result instanceof Throwable)
+                throw new AssertionError("An error occured on thread.", (Throwable) result);
+            assertTrue(isVirtual((Thread) thread));
+    
+            // Test scheduleAtFixedRate on virtual threads
+            final LinkedBlockingQueue<Object> resultsFixedRate = new LinkedBlockingQueue<>();
+            ScheduledFuture<?> future = virtualManagedScheduledExecutorAnno.scheduleAtFixedRate(
+                    new LookupActionCaptureThread(resultsFixedRate, "java:app/concurrent/ManagedScheduledExecutorAnnoVirtual"),
+                    0, TestConstants.pollInterval.toMillis(), TimeUnit.MILLISECONDS);
+    
+            assertTimeoutPreemptively(TestConstants.waitTimeout, () -> {
+                for (; 6 >= resultsFixedRate.size(); Wait.sleep(TestConstants.pollInterval)) {
+                    // Wait till we have at least 6 results indicating 3 runs of schedule
+                }
+            });
+    
+            Wait.waitCancelFuture(future); // Cancel execution of timer
+    
+            int resultsAfterCancel = resultsFixedRate.size();
+            assertTrue(resultsAfterCancel >= 6, "Should have executed schedule at least 3 times.");
+    
+            Thread thread0 = (Thread) resultsFixedRate.poll(1, TimeUnit.MILLISECONDS);
+            Object result0 = resultsFixedRate.poll(1, TimeUnit.MILLISECONDS);
+    
+            Thread thread1 = (Thread) resultsFixedRate.poll(1, TimeUnit.MILLISECONDS);
+            Object result1 = resultsFixedRate.poll(1, TimeUnit.MILLISECONDS);
+    
+            Thread thread2 = (Thread) resultsFixedRate.poll(1, TimeUnit.MILLISECONDS);
+            Object result2 = resultsFixedRate.poll(1, TimeUnit.MILLISECONDS);
+            
+            assertEquals(resultsAfterCancel - 6, resultsFixedRate.size(),
+                    "No more results should have been added to queue after task was cancelled");
+    
+            Set<Thread> virtualThreads = new HashSet<Thread>();
+    
+            if (result0 instanceof Throwable)
+                throw new AssertionError("An error occured on thread.", (Throwable) result0);
+            assertTrue(isVirtual((Thread) thread0));
```

Comment:
Similar to first comment. This is not a requirement, but very unlikely it would not be the case.